vcd/resource_vcd_vapp_vm.go

```diff
@@ -993,6 +1073,17 @@ func resourceVcdVAppVmUpdateExecute(d *schema.ResourceData, meta interface{}) er
 			}
 		}
 
+		if d.HasChange("network") && networksNeedToChange {
```

Comment:
This is becoming not understandable. You read this line and think:
- What's the difference between `d.HasChange("network")` and `networksNeedToChange`?
- Why we have "butter on butter" here?

Please, how can this be clarified? Maybe `networksNeedToChange` naming does not represent what it actually is?
Same holds for all three `...NeedToChange` variables (lines +992 to +994).

Reply:
There is a few cases. When field or network added then it can be changed with hot way, if network removed or initiated from`creation` then cold change should be used.
Other fields has other requirements or limitations.